actionpack/lib/action_dispatch/http/filter_parameters.rb

```diff
@@ -18,13 +18,18 @@ module Http
     #   env["action_dispatch.parameter_filter"] = [:foo, "bar"]
     #   => replaces the value to all keys matching /foo|bar/i with "[FILTERED]"
     #
+    #   env["action_dispatch.parameter_filter"] = [ /\Apin\z/i, /\Apin_/i ]
+    #   => replaces the value for the exact (case-insensitive) key 'pin' and all
+    #   (case-insensitive) keys beginning with 'pin_', with "[FILTERED]"
+    #   Does not match keys with 'pin' as a substring, such as 'shipping_id'.
+    #
     #   env["action_dispatch.parameter_filter"] = [ "credit_card.code" ]
     #   => replaces { credit_card: {code: "xxxx"} } with "[FILTERED]", does not
     #   change { file: { code: "xxxx"} }
     #
-    #   env["action_dispatch.parameter_filter"] = -> (k, v) do
+    #   env["action_dispatch.parameter_filter"] = [-> (k, v) do
```

Comment:
It works by `parameter_filter_for(Array(user_key) + ENV_MATCH)` as it is.

https://github.com/rails/rails/blob/924c31f8866892a3aedbd45a03b5f7e3ea367115/actionpack/lib/action_dispatch/http/filter_parameters.rb#L70-L78

Reply:
This is only true inside `#env_filter`. The call in `#parameter_filter` does not use an Array wrapper. I therefore experienced an exception in real code as a result of using the example given.

Equally, none of the otherwise identical examples at https://github.com/rails/rails/blob/main/activesupport/lib/active_support/parameter_filter.rb illustrate anything but an array argument, so this odd-one-out looks like an inconsistency.

Reply:
`Aarray()` for `parameter_filter` was lost in https://github.com/rails/rails/commit/55ae903c3fd2965f85a6e2e45ddd6b5b1156be91#diff-afe35a455271dbb6fafdde7329a0a9c8151cb060c0835a4812dfcdf4ef38bd5fL52-R67, it is considered as a bug.

Can you fix it instead of changing the doc?

Reply:
I understand the suggestion, but that would leave the interface dissimilar to ActiveSupport::ParameterFilter, which has a shared history and is usually instantiated from the same config. This feels to me like a POLA violation and isn't something I'm comfortable supporting as the PR author.

Instead, I'll back this out, and leave reviewing the difference in interfaces, and what to do about it, and the consequences of changing them, to folks who've been working on the Rails source longer than I have.